### DISCUSSION TITLE
Пересчет большой таблицы

### DISCUSSION THREAD

**Discussion by juramebel** [totumonline/totum-ru-issues-and-discussions#156] [Как это сделать?] [answered]
Здравствуйте.
Простая таблица. В ней номенклатура. Уже около 18 000 строк (так много, чтобы стандартные позиции пользователи не искали и не заносили сразу), то есть многие просто с кол-вом 0 на складе.
Много вычисляемых полей.
Не могу пересчитать больше 500 строк, упираюсь в ограничение памяти. 
Как можно выйти из ситуации?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Ну можно пересчитать руками, по 500 строк, но это муть (а иногда бывают таблицы по 100 млн строк и их нифига руками не пересчитаешь).

Для таких задач делается разовый Cron.

Нам потребуется поле, в которое мы будем записывать максимальный id до которого пересчитали. Я сделаю его прям в таблице Кронов — `h_store_num` (вы можете поставить его в любое другое место)

Создаем Cron:

```
a1= : reCalculate(table: 'you_table'; where: 'id' = $list)

	~list: selectList(table: 'you_table'; field: 'id'; where: 'id' > #h_store_num; limit: 10;  order: 'id' asc)

a2=: set(table: $#ntn; field: 'h_store_num' = $last_num)

last_num: $list[$count]

	count: listCount(list: $list) - 1

a3=: if(condition: $list = $#lc; then: $set_off)

	set_off: set(table: $#ntn; field: 'status' = false; where: 'id' = #id)

```

Из важного:

+ Сколько строк пересчитывать задается в `limit` функции `selectList`
+ `selectList` обязательно должен иметь `order`
+ строка с selectList обязательно должна быть зафиксирована `~`
+ Поставьте интервал Крона так, что бы они точно не пересекались с предыдущим (минимальный интервал у крона 1 минута, но может вам стоит сделать 5)
+ в a3 прописано отключение крона, когда он дошел до конца таблицы

**Reply by juramebel** [reply]
Супер) все работает.
Еще и видно благодаря h_store_num как далеко уже прошло.
